src/consentless/dynamic/liveblog-inline.ts

```diff
@@ -1,165 +0,0 @@
-const getWindowHeight = (doc = document) => {
-	if (doc.documentElement.clientHeight) {
-		return doc.documentElement.clientHeight;
-	}
-	return 0; // #? zero, or throw an error?
-};
-
-const getSpaceFillerRules = (
-	windowHeight: number,
-	shouldUpdate = false,
-): SpacefinderRules => {
-	let prevSlot: SpacefinderItem | undefined;
-
-	const isEnoughSpaceBetweenSlots = (
-		prevSlot: SpacefinderItem,
-		slot: SpacefinderItem,
-	) => Math.abs(slot.top - prevSlot.top) > windowHeight * AD_SPACE_MULTIPLIER;
-
-	const filterSlot = (slot: SpacefinderItem) => {
-		if (!prevSlot) {
-			prevSlot = slot;
-			return !shouldUpdate;
-		} else if (isEnoughSpaceBetweenSlots(prevSlot, slot)) {
-			prevSlot = slot;
-			return true;
-		}
-		return false;
-	};
-
-	return {
-		bodySelector: '.js-liveblog-body',
-		slotSelector: ' > .block',
-		fromBottom: shouldUpdate,
-		startAt: shouldUpdate ? firstSlot : undefined,
-		absoluteMinAbove: shouldUpdate ? 0 : WINDOWHEIGHT * PAGE_TOP_MULTIPLIER,
-		minAbove: 0,
-		minBelow: 0,
-		clearContentMeta: 0,
-		selectors: {},
-		filter: filterSlot,
-	};
-};
-
-const insertAdAtPara = (para: Node): Promise<void> => {
-	const container: HTMLElement = document.createElement('div');
-	container.className = `ad-slot-container`;
-	const name = `inline${AD_COUNTER + 1}`;
-	const adSlot = createAdSlot('inline', {
-		name,
-		classes: 'liveblog-inline',
-	});
-
-	container.appendChild(adSlot);
-
-	return fastdom
-		.mutate(() => {
-			if (para.parentNode) {
-				/* ads are inserted after the block on liveblogs */
-				para.parentNode.insertBefore(container, para.nextSibling);
-			}
-		})
-		.then(() => {
-			defineSlot(adSlot, name, 'inline');
-		});
-};
-
-const insertAds: SpacefinderWriter = async (paras) => {
-	const fastdomPromises = [];
-	for (let i = 0; i < paras.length && AD_COUNTER < MAX_ADS; i += 1) {
-		const para = paras[i];
-		if (para?.parentNode) {
-			const result = insertAdAtPara(para);
-			fastdomPromises.push(result);
-			AD_COUNTER += 1;
-		}
-	}
-	await Promise.all(fastdomPromises);
-};
-
-const onUpdate = () => {
-	// eslint-disable-next-line no-use-before-define -- circular reference
-	stopListening();
-	const rules = getSpaceFillerRules(WINDOWHEIGHT, true);
-	// eslint-disable-next-line no-use-before-define -- circular reference
-	void fill(rules);
-};
-
-const startListening = () => {
-	document.addEventListener('liveblog:blocks-updated', onUpdate);
-};
-
-const stopListening = () => {
-	document.removeEventListener('liveblog:blocks-updated', onUpdate);
-};
-
-const fill = (rules: SpacefinderRules) => {
-	const options: SpacefinderOptions = { pass: 'inline1' };
-
-	return spaceFiller.fillSpace(rules, insertAds, options).then(() => {
-		if (AD_COUNTER < MAX_ADS) {
-			const el = document.querySelector(
-				`${rules.bodySelector} > .ad-slot`,
-			);
-			if (el && el.previousSibling instanceof HTMLElement) {
-				firstSlot = el.previousSibling;
-			} else {
-				firstSlot = undefined;
-			}
-			startListening();
```

Comment:
Minor (probably lack of knowledge) question: can we retain the functionality here that adds slots when new blocks are added, just so we don't lose it? Then migrate it to DCR as a health ticket at some point?

Reply:
This is a task that still need to be done. It involves some refactoring to separate the code that inserts new blocks from the code that makes calls to GAM for new ads. I'll create a ticket for this follow-up work.